### PR TITLE
feat(codegen): add svelte support

### LINKED ISSUE #89: CodeGen: add Svelte support
Hi, are there any plans to support [Svelte](http://svelte.technology/)?

Thanks!

### PR DESCRIPTION
**Pull request by sumitparakh**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

## Other information

Closes #89

### CODE REVIEW

**Review by Jefiozie — COMMENTED** [review]
Did you update the NX library our are the changes around the eslint just new and we've not used this on other places? For the rest of the code, it looks good!

**Comment by sumitparakh** [comment]
@Jefiozie . Yeah.. I had latest version of nx installed on my system and I used following command to create node library:-

```
nx generate @nrwl/node:library <library-name>
```

After this, nx automatically added eslint config file along with its dependencies.. Should I keep them or remove them?

**Comment by Jefiozie** [comment]
@sumitparakh  ah that explains a couple of things. I already looked at it but couldn't find the reason why I was missing the angular builders in the `angular.json` file, but your explanation is clear. At this moment we use angular-libraries and not node libraries. Could you change the library to be an Angular lib instead of a node library? I'm sorry I've not seen this sooner.

**Comment by sumitparakh** [comment]
Ok. Will do that

**Comment by manekinekko** [comment]
@sumitparakh any updates about this PR?

File: .eslintrc
Comment (manekinekko) [inline .eslintrc:1]: is `eslint` really needed for this lib?

File: libs/svelte-codegen/.eslintrc
Comment (manekinekko) [inline libs/svelte-codegen/.eslintrc:1]: same as above

File: package.json
Comment (manekinekko) [inline package.json:112]: same as above

**Review by manekinekko — CHANGES_REQUESTED** [review]

**Comment by manekinekko** [comment]
Could you also [rebase your changes](https://github.com/xlayers/xlayers/blob/main/CONTRIBUTING.md#-submitting-a-pull-request-pr) (please do not merge from `main`)

**Comment by sumitparakh** [comment]
Hi @manekinekko , will finish it this weekend.

File: .eslintrc
Comment (sumitparakh) [reply .eslintrc]: Not needed. Removed the file

File: libs/svelte-codegen/.eslintrc
Comment (sumitparakh) [reply libs/svelte-codegen/.eslintrc]: Removed

File: package.json
Comment (sumitparakh) [reply package.json]: Removed.

**Comment by sumitparakh** [comment]
Hi @manekinekko & @Jefiozie , I've made my final changes. Please check now.

**Comment by sumitparakh** [comment]
> Could you also [rebase your changes](https://github.com/xlayers/xlayers/blob/main/CONTRIBUTING.md#-submitting-a-pull-request-pr) (please do not merge from `main`)

Hi @manekinekko, I was a bit confused by `do not merge from main`  line.. So, instead of taking pull from main in my feature branch, I just took rebase from main branch (`git rebase main -i`), as suggested in the guideline. I hope I did it right.

File: apps/xlayers/src/app/editor/code/editor-container/editor-container.component.ts
Comment (Jefiozie) [inline apps/xlayers/src/app/editor/code/editor-container/editor-container.component.ts:61]: I think this function should be removed and the onChange on [L45](https://github.com/xlayers/xlayers/pull/397/files#L45) should be used.

File: libs/svelte-codegen/.eslintrc.json
Comment (Jefiozie) [inline libs/svelte-codegen/.eslintrc.json:11]: Should this be renamed to `svelte-codegen`

**Review by Jefiozie — CHANGES_REQUESTED** [review]
Some minor improvements and then we are good to go.

File: apps/xlayers/src/app/editor/code/editor-container/editor-container.component.ts
Comment (sumitparakh) [reply apps/xlayers/src/app/editor/code/editor-container/editor-container.component.ts:61]: Done. Removed it because it was unused.

**Comment by sumitparakh** [comment]
> Some minor improvements and then we are good to go.

Done

**Review by Jefiozie — APPROVED** [review]
LGTM @manekinekko, would you mind doing a second review in case I missed something

**Review by manekinekko — APPROVED** [review]
Awesome job! Thank you 🎉

**Pull request merged**
